proto/usr/triton/bin/register-agent

```diff
@@ -0,0 +1,261 @@
+#!/bin/bash
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+#
+# Copyright 2021 Joyent, Inc.
+#
+
+#
+# We must load the SDC configuration before setting any strict error handling
+# options.
+#
+. /usr/triton/bin/config.sh
+load_sdc_config
+
+ROOT="$(cd "$(dirname "$0")"/../ 2>/dev/null && pwd)"
+
+. "${ROOT}/lib/error_handler.sh"
+. "${ROOT}/lib/trace_logger.sh"
+
+set -o nounset
+
+function usage
+{
+    echo "Usage: install-one-agent  AGENT_NAME  AGENT_DIR"
+    exit 1
+}
+
+if ! (( $# == 2 )); then
+    usage
+fi
+
+AGENT=$1
+AGENT_DIR=$2
+
+if [[ $AGENT == "-h" || $AGENT_DIR == "-h" ]]; then
+    usage
+fi
+
+if [[ ! -d $AGENT_DIR ]]; then
+    echo "Agent directory does not exist: $AGENT_DIR"
+    exit 1
+fi
+
+# Ur-agent does not get registered with SAPI.
+if [[ $AGENT == "ur-agent" ]]; then
+    echo "Ur-agent cannot be registered"
+    exit 1
+fi
+
+# SAPI details
+if [[ -z "${CONFIG_sapi_domain}" ]]; then
+    fatal '"sapi_domain" was not found in "node.config".'
+fi
+SAPI_URL="http://${CONFIG_sapi_domain}"
+
+ETC_DIR="/opt/smartdc/agents/etc"
```

Comment:
I'm not entirely sure why it's still in `/opt/smartdc`, but that's currently where it gets put. My assumption right now is that there's something in there that depends on that location, or maybe it was chosen to be compatible with SmartOS (e.g., for doing things like `sdc-oneachnode -a 'cat /opt/smartdc/agents/lib/node_modules/cn-agent/image_uuid'`).

Personally, I like being able to not have to distinguish paths when doing sdc-oneachnode, and I even need to make a change to illumos-joyent to include `/usr/triton/bin` so that `sdc-oneachnode -a 'vmadm list'` works.

With the project status being "technology preview" I think we've got flexibility here to try some things and optionally move stuff around later.

Reply:
I'll VERY much buy `sdc-oneachnode` for a reason why to keep that the way it is.